### ISSUE TITLE
Saved images cannot handle non-ASCII characters in filenames

### ISSUE THREAD

**Issue by andylizi** [Breakthrough/PySceneDetect#315]
### Bug Description

`save-images` filenames become gibberish if they contain non-ASCII characters.

![](https://user-images.githubusercontent.com/12008103/226336849-735d2529-974c-4dcd-9265-c30976c3caac.png)

### Reproduction Steps

1. On Windows, rename any video file to `ééé.mp4`, or any other non-ASCII name.

2. Run `scenedetect -i "ééé.mp4" detect-threshold save-images`

### Expected Behavior

Image files with the correct name should be generated, such as `ééé-Scene-001-01.jpg`

### Computing Environment

 - OS: Windows 11
 - Python Version: 3.11
 - OpenCV Version: 4.7.0

### Additional Information

This is caused by a known OpenCV bug.

https://github.com/Breakthrough/PySceneDetect/blob/384da8d698ce56a661ada53e5f52f706f718e840/scenedetect/scene_manager.py#L499

It was mentioned in this StackOverflow question: [OpenCv imwrite doesn't work because of special character in file path](https://stackoverflow.com/questions/44330084/opencv-imwrite-doesnt-work-because-of-special-character-in-file-path)

**Comment by wjs018** [comment]
This is a good edge case to bring attention to. I think it is beyond our scope to fix this issue upstream in OpenCV, but I can update our documentation to bring attention to this. I am guessing this only applies to images and not to saving split portions of the video since the video splitting isn't being done by OpenCV.

I will leave this issue open until I get a PR put together to update the docs. Thanks for bringing this to our attention @andylizi

**Comment by andylizi** [comment]
This OpenCV bug was discovered more than 5 years ago and still haven't been fixed (although there is opencv/opencv#21961), so perhaps it would be better to apply a workaround for now?

```python
success, im_buffer = cv2.imencode("." + image_extension, frame_im, imwrite_param)
im_buffer.tofile(get_and_create_path(file_path, output_dir))
```
